pkgs/applications/networking/browsers/chromium/upstream-info.json

```diff
@@ -1,5 +1,7 @@
 {
   "stable": {
+    "version_darwin": "99.0.4844.51",
+    "sha256_darwin": "14v5r6qn34k7jw2qilay4b3495gcjq701714732q23zn17jfynv6",
```

Comment:
That structure is a blocker for me.
1) It will cause merge conflicts (Git doesn't know enough about the structure by default)
2) It should at least have a structure like `stable.{linux,darwin}.*` (vs. the `_darwin` suffix). and for Chrome we currently use `sha256bin64`.

Actually I think it'd be best to add a completely new file (maybe something like `darwin-sources.json`) for Darwin. It might be useful to re-use the update script but we should avoid too much Darwin-specific complexity (currently it requires some major changes and additional dependencies - and the script is already more complex than it should be).

Also: Will you be up to maintain the macOS versions (i.e., subscribing to release notifications (e.g., via the RSS feed) and updating within days)? Because I won't have time/motivation for it and updates are security critical (therefore you also have to decide if you want to maintain all three channels or just the stable channel). - That's actually the most important question for now.

Reply:
> Are you suggesting to have a separate update script too?

Not sure, depends on the final implementation (if lots of changes are required or if we can re-use most of the code).

> It seems like it should be possible to do with GitHub actions.

Yes. I actually have this on my TODO list for a while but never got around to it as it lacks priority (I doubt it would help that much and a lot of boring code would probably have to be written for the testing, to avoid duplicate PRs, etc.).

> Isn't this a general problem? I'm surprised nixpkgs doesn't have a solution to this already.

We have a bot that can update some packages if the build still succeeds but we don't have a good and generalized solution yet (the problem are some details / exotic packages / etc.).

> Whoever runs the update will be updating both Darwin and Linux at the same time (that's what I have been doing and since you have already updated Linux then those are left as-is).

That would be ideal. Unfortunately the Darwin channel doesn't always match the Linux channel though (but it might not be that bad - I only noticed some extra updates to the Darwin channel so far).

> I think this makes the most sense as the versions will be in sync across each platform.

There's no need for `version_darwin` if that would be the case.

Anyway, just some quick notes, we should figure out the impurity issues first.